PublicMapping/district-builder-2
# Adapt ingestion script to output static data files + ADR

## Overview

This PR modifies the `process-geojson` script to output static data for demographic and geolevel data. A `static-metadata.json` file is also written out which provides details about the files that are output (notably `bytesPerElement`, which specifies what kind of typed array is used for each file). A small ADR was also added which discusses the process.

### Demo

![db-static-ls](https://user-images.githubusercontent.com/6386/76805897-6b4ae280-67b6-11ea-8fc6-532808baf4bb.png)

![db-static-html](https://user-images.githubusercontent.com/6386/76805945-84539380-67b6-11ea-8777-3e364bb2d4c9.png)

### Notes

I tested this successfully with each of Delaware, Pennsylvania, and California GeoJSON files.

## Testing Instructions

 * cd to `src/manage`
 * Make a directory for output, e.g. `mkdir output`
 * Run the script pointing to your DE.geojson test file, e.g.: `./bin/run process-geojson ~/DE.geojson -l block,tract,county -o output`
 * It should run successfully. Take a look at the output to verify.
 * If you'd like to verify this data can be loaded up to the client-side, you can store the following contents in `index.html`, serve that directory (with something like `python2 -m SimpleHTTPServer`), load it up and take a look at the console. It should match the screenshot above, and I verified that the information in that screenshot is correct.
```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
  </head>
  <script>
    fetch("output/population.buf")
      .then(res => res.arrayBuffer())
      .then(data => console.log("population: ", new Uint16Array(data)));
  </script>
</html>
```

Closes #9 
Closes #35
